[DIR-197] new services page

stefan-kracht opened the pull request:
This PR implements the services page in the new UI

### [Preview 🚀](https://direktiv-ui-refs-pull-446-merge.direktiv.dev/stefan/explorer/tree)

I added a description of how it works and a lot of possible test cases in the [E2E Tests Ticket](https://linear.app/direktiv/issue/DIR-757/services-page).

**!! The API is not very reliable currently, some things might not work properly and could not finally tested in detail. I added a [ticket](https://linear.app/direktiv/issue/DIR-758/services-page-needs-more-testing-when-api-works-reliable) for that !!** 

### Notes

- The component for creating a service and a revision are very similar and could technically be merged into one. But at the moment both have a pretty different prop interface and since the mutation and query hooks can not be called conditionally these hooks must be hoisted, which currently adds a level of complexity. So a future abstraction could be to somehow only abstract the visual parts (the form) into a common component. 

### Other things out of services scope

- updated the `.vscode` folder rule in `.gitgnore` (ignore all, except `.vscode/extensions.json`) to allow other extensions to use that folder and not track it in this repo
- updated the `FormErrors` component to handle an empty `key` string (`src/pages/namespace/Services/Detail/Create.tsx` adds a general error message that is not linked to a certain field)
- Made popover in `ChildInstances` wider to support longer names
- Added an `ErrorPage` to all routes
- Added the possibility to use our Log component without timestamps
- `NoResults` component
  - Added Button support (and used it in empty directories)
  - Made Icon bigger and light gray
  - Added an Icon to the `NoResult` instance in the Variables-, Secrets- and Registries-Component
- updated `yarn run check` to run all checks in parallel (saves 10 seconds, and has the advantage of always seeing all check results and not the results until the point of the first one failing)
- adjusted one test after the new API was deployed (4f7dfd9792ca010e47d6a9b7ca2c3a3083fd46bd). The shape of a server result seems to have changed (for the better), no breaking change.
- Made Run Workflow button responsive <details><summary>show screenshots</summary> 
  Before
  ![CleanShot 2023-08-23 at 14 08 40](https://github.com/direktiv/direktiv-ui/assets/121789579/1ff63b83-f291-4fac-be74-d200869ca7af)
  After
  ![CleanShot 2023-08-23 at 14 08 14](https://github.com/direktiv/direktiv-ui/assets/121789579/103c87d0-1d90-4faf-85bc-1cb04fdd6f3d)
</details> 


### Follow-Up Tickets
- [E2E Tests](https://linear.app/direktiv/issue/DIR-757/services-page)
- [Services page needs more testing when API works reliable](https://linear.app/direktiv/issue/DIR-758/services-page-needs-more-testing-when-api-works-reliable)

sebxian reviewed (changes requested):
The services page and sub pages look good and seem to work as they should.

I must admit, however, that with my limited knowledge and limited time I can invest right now this is only a somewhat superficial impression. It would be very important to have an expert look at this as soon as possible and test it more thoroughly (after merging?).

I have made some suggestions in the code, mostly regarding spelling/readability.

stefan-kracht commented on src/api/services/__tests__/schema.test.ts:
```diff
@@ -0,0 +1,59 @@
+      expect(serviceNameSchema.safeParse("fOldername").success).toBe(false);
+    });
+
+    test("can not end with various characters that are not lowercase characters or a digit", () => {
```
I added 1 😊  6440dd7ae57d403b8cec9a6c337e450031503f31

stefan-kracht commented on src/api/services/query/getAll.ts:
```diff
@@ -0,0 +1,129 @@
+    urlParams: { namespace },
+  }).then((res) => ({
+    // TODO: this should be changed in the backend
+    // reverse the order of functions (newer first)
```
Yes, this is still an issue I wrote a [ticket](https://linear.app/direktiv/issue/DIR-784/change-order-of-services) and mentioned it in the comment (0f2341e49d4234567bc751c132e863b3ae7416f8)

stefan-kracht commented on src/pages/namespace/Services/Detail/Create.tsx:
```diff
@@ -0,0 +1,224 @@
+          return true;
+        },
+        {
+          message: t("pages.services.revision.create.noChanges"),
```
I like that idea, we should definitely introduce that

stefan-kracht commented on src/pages/namespace/Services/Detail/Create.tsx:
```diff
@@ -0,0 +1,224 @@
+          message: t("pages.services.revision.create.noChanges"),
+        }
+        /**
+         * when no default values are available, it problably means that
+         * there is no previous revision. In this case, we don't let the
+         * user create one
```
Yes, we could phrase it like that (1b3a96473fad988b4d52eb873f4c03f46fb108ee)

stefan-kracht commented on src/pages/namespace/Explorer/Tree/NoResult.tsx:
```diff
@@ -22,48 +23,52 @@ const NoResult: FC = () => {
+    <div className="flex flex-col items-center gap-y-5">
+      <NoResultContainer
+        icon={FolderOpen}
+        button={
```
Very good idea (3854b37f513d2969f745c448c09ca60770d458f3)!

stefan-kracht commented on src/componentsNext/FormErrors/index.tsx:
```diff
@@ -10,12 +10,12 @@ type FormErrorsProps = ComponentProps<typeof Alert> & { errors: errorsType };
         {entries.map(([key, value]) => (
-          <li key={key}>{`${key}: ${
+          // note: key might also be an empty string
+          <li key={key}>{`${key && `${key} :`} ${
```
Yes, that's a very good catch! (035e178ea2e8aff014e67bfd68b2e02d0ef772cc)

stefan-kracht commented on src/api/services/schema/revisions.ts:
```diff
@@ -0,0 +1,197 @@
+    reason: z.string(),
+    message: z.string(),
+  })
+  // handle status: False with reason: NoTraffic as status: True
```
Yes, this was not very well explained, I hope this is better: 566804dbb7f98f9a77660191c4104fefdc8182a9

stefan-kracht commented on src/api/services/query/getDetails.ts:
```diff
@@ -0,0 +1,157 @@
+         * when streaming. The streaming payload has some minor inconsistencies
+         * with the revision schema. However, the fields that we keep from the
+         * cache are long living oney like the creation date and the name (which
+         * acts like an id)
```
fixed in d0fa1742611e6a9894d39d4128ccf3cc034ace02

stefan-kracht commented:
Thanks for your review @sebxian. I somehow have to reduce the need of language cop in the future :D

- [one minor unresolved issue, that you could check](https://github.com/direktiv/direktiv-ui/pull/446/files/4f7dfd9792ca010e47d6a9b7ca2c3a3083fd46bd#r1307692524) 
- I also 
  - merged the base branch back in and fixed a merge conflict
  - updated an unrelated test according to your language suggestions (6afe1f7e6cd3768b79a58dfbe0d0970166c5c58e)

sebxian reviewed (approved):
Thanks for the updates, @stefan-kracht, they all look good to me!
